### PR TITLE
[knowledge-store]: Use more generic types in API

### PR DESCRIPTION
**Pull request by cbornet**

### CODE REVIEW

File: libs/knowledge-store/ragstack_knowledge_store/base.py [hunk 1]
```diff
@@ -113,13 +113,19 @@ async def aadd_nodes(
...
         """Add nodes to the knowledge store
 
         Args:
             nodes: the nodes to add.
         """
-        return await run_in_executor(None, self.add_nodes, nodes, **kwargs)
+        iterator = iter(await run_in_executor(None, self.add_nodes, nodes, **kwargs))
+        done = object()
+        while True:
+            doc = await run_in_executor(None, next, iterator, done)
```
Comment (bjchambers) [inline libs/knowledge-store/ragstack_knowledge_store/base.py:125]: Should this use `aiter` and `anext`?

File: libs/knowledge-store/ragstack_knowledge_store/cassandra.py [hunk 2]
```diff
@@ -512,7 +512,7 @@ def similarity_search_by_vector(
     def _query_by_ids(
         self,
         ids: Iterable[str],
-    ) -> Iterable[Document]:
+    ) -> List[Document]:
```
Comment (bjchambers) [inline libs/knowledge-store/ragstack_knowledge_store/cassandra.py:515]: Should these stick with `Iterable`? Wouldn't that let them change implementation more easily?

**Review by bjchambers — APPROVED** [review]

File: libs/knowledge-store/ragstack_knowledge_store/cassandra.py (hunk 2 above)
Comment (cbornet) [reply libs/knowledge-store/ragstack_knowledge_store/cassandra.py:515]: It's an internal method so the signature can be changed later if needed.

File: libs/knowledge-store/ragstack_knowledge_store/base.py (hunk 1 above)
Comment (cbornet) [reply libs/knowledge-store/ragstack_knowledge_store/base.py:125]: `aiter` and `anext` work on `AsyncIterable`. Here `iterator` is an `Iterable`.

File: libs/knowledge-store/ragstack_knowledge_store/cassandra.py (hunk 2 above)
Comment (cbornet) [reply libs/knowledge-store/ragstack_knowledge_store/cassandra.py]: Yes, it 's better to keep `Iterable` to evolve it more easily.
Done.

**Pull request merged**